### PR TITLE
CPM-1068: Update attribute normalization

### PR DESCRIPTION
**Pull request by Beninho**
**Description (for Contributor and Core Developer)**

- add “is_main_identifier“ field (only for identifier attributes)
- exposed in API, not in exports
- the value can not be imported

**Definition Of Done (for Core Developer only)**

- [X] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Resources/config/updaters.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ services:
             - '@pim_catalog.registry.attribute_type'
             - '@pim_structure.updater.translatable'
             - ['auto_option_sorting', 'is_read_only', 'default_value']
-            - ['group_labels']
+            - ['group_labels', 'is_main_identifier']
```
Comment (Beninho) [inline src/Akeneo/Pim/Structure/Bundle/Resources/config/updaters.yml:33]: update just for api

File: src/Akeneo/Tool/Bundle/VersioningBundle/tests/integration/Normalizer/Flat/AttributeIntegration.php [hunk 2]
```diff
@@ -47,6 +47,7 @@ public function testAttributeIdentifier()
             'options'                => null,
             'scope'                  => 'Global',
             'required'               => true,
+            'is_main_identifier'     => true,
```
Comment (Beninho) [inline src/Akeneo/Tool/Bundle/VersioningBundle/tests/integration/Normalizer/Flat/AttributeIntegration.php:50]: remove this

File: src/Akeneo/Pim/Structure/Bundle/Application/MainIdentifier/ChangeMainIdentifierHandler.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Structure\Bundle\Application\MainIdentifier;
```
Comment (cemorin) [inline src/Akeneo/Pim/Structure/Bundle/Application/MainIdentifier/ChangeMainIdentifierHandler.php:5]: TODO: remove from PR

File: src/Akeneo/Pim/Structure/Bundle/Controller/InternalApi/MainIdentifierController.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+namespace Akeneo\Pim\Structure\Bundle\Controller\InternalApi;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class MainIdentifierController
```
Comment (cemorin) [inline src/Akeneo/Pim/Structure/Bundle/Controller/InternalApi/MainIdentifierController.php:14]: remove

File: src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/AttributeRepository.php [hunk 5]
```diff
@@ -375,4 +376,9 @@ public function findAttributesByFamily(FamilyInterface $family)
 
         return $qb->getQuery()->getResult();
     }
+
+    public function updateMainIdentifier(AttributeInterface $attribute): void
```
Comment (cemorin) [inline src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/AttributeRepository.php:380]: remove

File: tests/legacy/features/pim/structure/attribute/export/export_attributes_csv.feature [hunk 6]
```diff
@@ -4,7 +4,7 @@ Feature: Export attributes
   As a product manager
   I need to be able to export attributes
 
-  @ce @critical
```
Comment (cemorin) [inline tests/legacy/features/pim/structure/attribute/export/export_attributes_csv.feature:7]: Do we really want it in EE?

**Review by mmetayer — APPROVED** [review]

**Pull request merged**